### PR TITLE
[BOOST-4842] feat(sdk): add gas used setting for variable incentives sdk helpers

### PR DESCRIPTION
**Pull request by Quazia**
🚨 Please review the [guidelines for contributing](https://github.com/boostxyz/boost-protocol/blob/main/.github/CONTRIBUTING.md) to this repository.

### Description
Similar to the other cheat codes we're using, uses max uint8 to indicate gas rebate for a variable criteria incentive.

💔 Thank you!

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fe4fd087915611012ad3bd2d6c6a31540d318725
</p>

File: packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts
Comment (sammccord) [inline packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts:1]: if we keep up this whole 255, 254, 253 field indexes as enums for special validation behavior, at some point we should export an enum. not saying this is a right now thing just musing

**Review by sammccord — APPROVED** [review]

File: packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts
Comment (Quazia) [reply packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts:1]: I was thinking that when I was writing it. I'm in favor of making that change in this PR for all the cheat codes. Good to establish now I reckon, enforces legibility down the line.

Comment (jonathandiep) [reply packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts:1]: +1 to this! I also get confused cause we reserve 255 for multiple things and we need to make it really clear (255 for incentive criteria, 255 for event action criteria, etc.)

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0a454b94b936f056a4b01e67d491ea32b2ce2888
</p>

File: packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts
Comment (Quazia) [reply packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts:1]: Hows this look?

**Review by sammccord — APPROVED** [review]
<3

**Pull request merged**